offsite-activity-signup/src/Code.js

```diff
@@ -0,0 +1,458 @@
+    return obj;
+  }, {});
+  for (var i = 0; i < numActivitiesPerPerson; ++i) {
+    var randomizedAttendees = shuffleArray_(attendees);
+    randomizedAttendees.forEach(function(attendee) {
+      makeChoice_(attendee, activitiesById);
+    });
+  }
+}
+
+/**
+ * Attempt to assign an activity for an attendee based on their preferences
+ * and current schedule.
+ *
+ * @param {object} attendee - Attendee looking to join an activity
+ * @param {object} activitiesById - Map of all available activities
+ */
+function makeChoice_(attendee, activitiesById) {
+  for (var i = 0; i < attendee.preferences.length; ++i) {
+    var activity = activitiesById[attendee.preferences[i]];
+    if (!activity) {
+      continue;
+    }
+    var canJoin = checkAvailability_(attendee, activity);
+    if (canJoin) {
+      attendee.assigned.push(activity);
+      activity.roster.push(attendee);
+      break;
+    }
+  }
+}
+
+/**
+ * Checks that an activity has capacity and doesn't conflict with
+ * previously assigned activities.
+ *
+ * @param {object} attendee - Attendee looking to join the activity
+ * @param {object} activity - Proposed activity
+ * @return {boolean} - True if attendee can join the activity
+ */
+function checkAvailability_(attendee, activity) {
+  if (activity.capacity <= activity.roster.length) {
+    return false;
+  }
+  var timesConflict = attendee.assigned.some(function(assignedActivity) {
+    return !(assignedActivity.startAt.getTime() > activity.endAt.getTime() ||
+      activity.startAt.getTime() > assignedActivity.endAt.getTime());
+  });
+  return !timesConflict;
+};
+
+/**
+ * Populates a sheet with the assigned activities for each attendee.
+ *
+ * @param {Spreadsheet} ss - Spreadsheet to write to.
+ * @param {object[]} attendees - Array of attendees with their activity assignments
+ */
+function writeAttendeeAssignments_(ss, attendees) {
+  var sheet = findOrCreateSheetByName_(ss, 'Activities by person');
+  sheet.clear();
+  sheet.appendRow(['Email address', 'Activities']);
+  sheet.getRange('B1:1').merge();
+  var rows = attendees.map(function(attendee) {
+    // Prefill row to ensure consistent length otherwise
+    // can't bulk update the sheet with range.setValues()
+    var row = fillArray_([], ACTIVITIES_PER_PERSON + 1, '');
+    row[0] = attendee.email;
+    attendee.assigned.forEach(function(activity, index) {
+      row[index + 1] = activity.description;
+    });
+    return row;
+  });
+  bulkAppendRows_(sheet, rows);
+  sheet.setFrozenRows(1);
+  sheet.getRange('1:1').setFontWeight('bold');
+  sheet.autoResizeColumns(1, sheet.getLastColumn());
+}
+
+/**
+ * Populates a sheet with the rosters for each activity.
+ *
+ * @param {Spreadsheet} ss - Spreadsheet to write to.
+ * @param {object[]} activities - Array of activities with their rosters
+ */
+function writeActivityRosters_(ss, activities) {
+  var sheet = findOrCreateSheetByName_(ss, 'Activity rosters');
+  var sheet = ss.getSheetByName(sheetName);
```

Comment:
I think you forgot to delete the old find or create code in this function.

Reply:
d'oh!